### ISSUE TITLE
Is neo4j-rake_tasks useful for production?

### ISSUE THREAD

**Issue by ProGM** [neo4jrb/neo4j#1160]
I'm doing a few benchmarks for my app, using [`derailed`](https://github.com/schneems/derailed_benchmarks) gem.

I noticed that `neo4j` is including `neo4j-rake_tasks`, that is useful for development, but I don't think it's very useful for production environment. Is there any way to exclude it?

I tried:

```ruby
gem 'neo4j-rake_tasks', require: false, group: :production
```

But... well. It was not a great idea :laughing:

**Comment by cheerfulstoic** [comment]
Can you just do this?

```ruby
gem 'neo4j-rake_tasks', group: [:development, :test]
```

Bundler should only auto-require for the environment's gems, I think

**Comment by ProGM** [comment]
I tried, but derailed says that it's still required. :\
I think bundler is ignoring me, since `neo4j-rake_tasks` is an explicit dipendence ...

**Comment by cheerfulstoic** [comment]
Oh, right, yeah, it probably is because `neo4j-core` depends on it...

Perhaps we should make `neo4j-rake_tasks` be an explicit dependency of apps.  We could remove it as a dependency of `neo4j-core`, though we'd probably want to give a deprecation warning first.  And/or maybe when we remove it we could temporarily create rake tasks in `neo4j-core` (which `neo4j-rake_tasks` would override) which just output a message about how you now need to require the gem explicitly and providing an example of how to do it just in `development` / `test`

Pinging @subvertallchris for an additional opinion

**Comment by thefliik** [comment]
I think closed by https://github.com/neo4jrb/neo4j-core/commit/4ac9a4156eaad06b3b11c29f7d6a33d1c4c1683c

**Comment by cheerfulstoic** [comment]
Yep, absolutely, thanks @thefliik !  `neo4j-rake_tasks` is no longer a depedency of `neo4j-core`

As @ProGM said, `neo4j-rake_tasks` is not a good idea for production (and that's mentioned in the docs in at least a couple of places, I think).  It's primarily for development and test, though in the past six months Neo4j, Inc has released the Neo4j Desktop app which is a great way to manage multiple Neo4j servers, so I'm starting to think that `neo4j-rake_tasks` should be deprecated entirely.
